### DISCUSSION TITLE
KeyWalletTouch rejected even though it works(Project ID: c83492abbe05a38fb9a0d366a9b5e047)

### DISCUSSION THREAD

**Discussion by DongOkSeo** [WalletConnect/walletconnect-monorepo#3519] [Developer Support] [answered]
I applied for wallet app "KeyWallet Touch" approval, but it was rejected due to "Wallet Testing." Afterward, I tested it at https://lab.web3modal.com/ and confirmed that it functions properly, working well with other Dapps. I kindly request a re-evaluation. 

reject message
Your update has been rejected. Reason:
"Wallet Testing - Could not verify if testing has been done on this wallet. As we do not have available V2 metric data for Sign. Please test your integration with our test dapp and submit for approval if successful. Link: https://lab.web3modal.com/"

**Comment by glitch-txs** [comment, marked as answer]
Hi, we need you to test this on labs as we are tracking this dapp for confirmation. Kindly test on labs and request again the verification 🙏
